internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+		},
+
+		"nfsv3_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+			ForceNew: true,
+		},
+
+		"allow_blob_public_access": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"shared_access_key_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"network_rules": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"bypass": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(storage.AzureServices),
+								string(storage.Logging),
+								string(storage.Metrics),
+								string(storage.None),
+							}, true),
+						},
+						Set: pluginsdk.HashString,
+					},
+
+					"ip_rules": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validate.StorageAccountIpRule,
+						},
+						Set: pluginsdk.HashString,
+					},
+
+					"virtual_network_subnet_ids": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Computed: true,
+						Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+						Set:      pluginsdk.HashString,
+					},
+
+					"default_action": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(storage.DefaultActionAllow),
+							string(storage.DefaultActionDeny),
+						}, false),
+					},
+
+					"private_link_access": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"endpoint_resource_id": {
+									Type:         pluginsdk.TypeString,
+									Required:     true,
+									ValidateFunc: azure.ValidateResourceID,
+								},
+
+								"endpoint_tenant_id": {
+									Type:         pluginsdk.TypeString,
+									Optional:     true,
+									Computed:     true,
+									ValidateFunc: validation.IsUUID,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"identity": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Computed: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"type": {
+						Type:             pluginsdk.TypeString,
+						Required:         true,
+						DiffSuppressFunc: suppress.CaseDifference,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(storage.IdentityTypeSystemAssigned),
+							string(storage.IdentityTypeSystemAssignedUserAssigned),
+							string(storage.IdentityTypeUserAssigned),
+						}, true),
```

Comment:
fwiw you don't need these for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```